tests/k8s/test_client.py

```diff
@@ -50,6 +50,10 @@ def url(self):
     def explicit_timeout(self):
         return 60
 
+    def test_session_factory(self):
+        session = _session_factory()
+        assert session is not None
```

Comment:
Is it possible to test that the `HTTPAdapter` is mounted for `http` and `https`? As it is, this test seems a bit pointless. It should check the relevant features of the session, and should probably have a name to reflect what it is testing.

`test_session_configured_for_retry` or something like that perhaps?

Reply:
I've updated the test with 4777d97